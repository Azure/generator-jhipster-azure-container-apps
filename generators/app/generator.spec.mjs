import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'app';
const BLUEPRINT_NAMESPACE = `jhipster:${SUB_GENERATOR}`;

describe('SubGenerator app of azure-container-apps JHipster blueprint', () => {
  describe('run', () => {
    let result;
    beforeAll(async function () {
      result = await helpers
        .create(BLUEPRINT_NAMESPACE)
        .withOptions({
          reproducible: true,
          defaults: true,
          baseName: 'jhipster',
          ignoreNeedlesError: true,
          blueprint: 'azure-container-apps',
        })
        .withJHipsterLookup()
        .run();
    });

    it('should succeed', () => {
      expect(result.getStateSnapshot()).toMatchSnapshot();
    });
  });
});
