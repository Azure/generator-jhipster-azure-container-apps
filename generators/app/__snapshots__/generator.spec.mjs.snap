// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator app of azure-container-apps JHipster blueprint > run > should succeed 1`] = `
{
  ".devcontainer/devcontainer.json": {
    "stateCleared": "modified",
  },
  ".gitattributes": {
    "stateCleared": "modified",
  },
  ".github/workflows/azure-dev.yml": {
    "stateCleared": "modified",
  },
  ".github/workflows/terraform/azure-dev.yml": {
    "stateCleared": "modified",
  },
  ".gitignore": {
    "stateCleared": "modified",
  },
  ".vscode/extensions.json": {
    "stateCleared": "modified",
  },
  ".vscode/launch.json": {
    "stateCleared": "modified",
  },
  ".vscode/tasks.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "LICENSE": {
    "stateCleared": "modified",
  },
  "OPTIONAL_FEATURES.md": {
    "stateCleared": "modified",
  },
  "README.md": {
    "stateCleared": "modified",
  },
  "assets/resources-with-apim.png": {
    "stateCleared": "modified",
  },
  "assets/resources.png": {
    "stateCleared": "modified",
  },
  "assets/urls.png": {
    "stateCleared": "modified",
  },
  "assets/web.png": {
    "stateCleared": "modified",
  },
  "azure.yaml": {
    "stateCleared": "modified",
  },
  "infra/abbreviations.json": {
    "stateCleared": "modified",
  },
  "infra/app/api.bicep": {
    "stateCleared": "modified",
  },
  "infra/app/apim-api-policy.xml": {
    "stateCleared": "modified",
  },
  "infra/app/apim-api.bicep": {
    "stateCleared": "modified",
  },
  "infra/app/db.bicep": {
    "stateCleared": "modified",
  },
  "infra/app/web.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/ai/cognitiveservices.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/ai/hub-dependencies.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/ai/hub.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/ai/project.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/config/configstore.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/cosmos-account.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/mongo/cosmos-mongo-account.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/mongo/cosmos-mongo-db.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/sql/cosmos-sql-account.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/sql/cosmos-sql-db.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/sql/cosmos-sql-role-assign.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/cosmos/sql/cosmos-sql-role-def.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/mysql/flexibleserver.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/postgresql/flexibleserver.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/database/sqlserver/sqlserver.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/gateway/apim.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/ai-environment.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/aks-agent-pool.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/aks-managed-cluster.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/aks.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/appservice-appsettings.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/appservice.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/appserviceplan.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/container-app-upsert.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/container-app.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/container-apps-environment.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/container-apps.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/container-registry.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/functions.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/host/staticwebapp.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/monitor/applicationinsights-dashboard.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/monitor/applicationinsights.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/monitor/loganalytics.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/monitor/monitoring.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/networking/cdn-endpoint.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/networking/cdn-profile.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/networking/cdn.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/search/search-services.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/aks-managed-cluster-access.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/configstore-access.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/keyvault-access.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/keyvault-secret.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/keyvault.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/registry-access.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/security/role.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/storage/storage-account.bicep": {
    "stateCleared": "modified",
  },
  "infra/core/testing/loadtesting.bicep": {
    "stateCleared": "modified",
  },
  "infra/main.bicep": {
    "stateCleared": "modified",
  },
  "infra/main.parameters.json": {
    "stateCleared": "modified",
  },
  "src/api/.mvn/wrapper/maven-wrapper.jar": {
    "stateCleared": "modified",
  },
  "src/api/.mvn/wrapper/maven-wrapper.properties": {
    "stateCleared": "modified",
  },
  "src/api/.openapi-generator/FILES": {
    "stateCleared": "modified",
  },
  "src/api/.openapi-generator/VERSION": {
    "stateCleared": "modified",
  },
  "src/api/.openapi-generator/openapi.yaml-default.sha256": {
    "stateCleared": "modified",
  },
  "src/api/Dockerfile": {
    "stateCleared": "modified",
  },
  "src/api/README.md": {
    "stateCleared": "modified",
  },
  "src/api/mvnw": {
    "stateCleared": "modified",
  },
  "src/api/mvnw.cmd": {
    "stateCleared": "modified",
  },
  "src/api/openapi.yaml": {
    "stateCleared": "modified",
  },
  "src/api/package-lock.json": {
    "stateCleared": "modified",
  },
  "src/api/package.json": {
    "stateCleared": "modified",
  },
  "src/api/pom.xml": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/SimpleTodoApplication.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/api/ApiUtil.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/api/ItemsApi.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/api/ListsApi.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/configuration/RFC3339DateFormat.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/configuration/StringToTodoStateConverter.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/configuration/WebConfiguration.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/controller/TodoItemsController.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/controller/TodoListsController.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/model/TodoItem.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/model/TodoList.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/model/TodoState.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/repository/TodoItemRepository.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/java/com/mycompany/myapp/repository/TodoListRepository.java": {
    "stateCleared": "modified",
  },
  "src/api/src/main/resources/application-cloud.properties": {
    "stateCleared": "modified",
  },
  "src/api/src/main/resources/application-local.properties": {
    "stateCleared": "modified",
  },
  "src/api/src/main/resources/application.properties": {
    "stateCleared": "modified",
  },
  "src/api/src/main/resources/applicationinsights.json": {
    "stateCleared": "modified",
  },
  "src/api/src/test/java/com/mycompany/myapp/SimpleTodoApplicationTests.java": {
    "stateCleared": "modified",
  },
  "src/web/.dockerignore": {
    "stateCleared": "modified",
  },
  "src/web/.eslintrc.cjs": {
    "stateCleared": "modified",
  },
  "src/web/Dockerfile": {
    "stateCleared": "modified",
  },
  "src/web/README.md": {
    "stateCleared": "modified",
  },
  "src/web/index.html": {
    "stateCleared": "modified",
  },
  "src/web/nginx/nginx.conf": {
    "stateCleared": "modified",
  },
  "src/web/package-lock.json": {
    "stateCleared": "modified",
  },
  "src/web/package.json": {
    "stateCleared": "modified",
  },
  "src/web/public/favicon.ico": {
    "stateCleared": "modified",
  },
  "src/web/public/manifest.json": {
    "stateCleared": "modified",
  },
  "src/web/src/@types/window.d.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/App.css": {
    "stateCleared": "modified",
  },
  "src/web/src/App.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/actions/actionCreators.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/actions/common.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/actions/itemActions.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/actions/listActions.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/components/telemetry.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/components/telemetryContext.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/components/telemetryWithAppInsights.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/components/todoContext.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/components/todoItemDetailPane.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/components/todoItemListPane.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/components/todoListMenu.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/config/index.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/index.css": {
    "stateCleared": "modified",
  },
  "src/web/src/index.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/layout/header.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/layout/layout.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/layout/sidebar.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/models/applicationState.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/models/index.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/models/todoItem.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/models/todoList.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/pages/homePage.tsx": {
    "stateCleared": "modified",
  },
  "src/web/src/react-app-env.d.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/reducers/index.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/reducers/listsReducer.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/reducers/selectedItemReducer.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/reducers/selectedListReducer.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/reportWebVitals.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/services/itemService.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/services/listService.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/services/restService.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/services/telemetryService.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/setupTests.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/ux/styles.ts": {
    "stateCleared": "modified",
  },
  "src/web/src/ux/theme.ts": {
    "stateCleared": "modified",
  },
  "src/web/tsconfig.json": {
    "stateCleared": "modified",
  },
  "src/web/tsconfig.node.json": {
    "stateCleared": "modified",
  },
  "src/web/vite.config.ts": {
    "stateCleared": "modified",
  },
  "src/web/web.config": {
    "stateCleared": "modified",
  },
}
`;
